beaconChain/helpers/stateTransitionHelpers.ts

```diff
@@ -290,10 +297,10 @@ function getCrosslinkCommitteesAtSlot(state: BeaconState, slot: SlotNumber, regi
     }
   }
 
-  const shuffling = getShuffling(seed, state.validatorRegistry, shufflingEpoch);
-  const offset = slot % EPOCH_LENGTH;
+  const shuffling: ValidatorIndex[][] = getShuffling(seed, state.validatorRegistry, shufflingEpoch);
+  const offset: int = slot.umod(new BN(EPOCH_LENGTH)).toNumber();
   const committeesPerSlot = Math.floor(committeesPerEpoch / EPOCH_LENGTH);
-  const slotStartShard = (shufflingStartShard + committeesPerSlot * offset) % SHARD_COUNT;
+  const slotStartShard = shufflingStartShard.addn(committeesPerSlot * offset).umod(new BN(SHARD_COUNT));
```

Comment:
should this ``committeesPerSlot * offset``  us BN arithmetic?

Reply:
I don't think it needs to use BN
they're both of type int and both have pretty low max values.

Reply:
Thought: should we just stay consistent and use BN everywhere?

Reply:
I tend to not think so, or at least not in this PR.
In this PR, we've used BN everywhere we "need" to, as defined by the spec. (eg: numbers specified as uint64 are BNs, functions changed to match that, nothing more)
Its more than just this line of code and the other change you requested below that would be needed to convert fully to bignumber/BN, and I think its worth doing it all consistently across the board if we want to go there.

Personally, I'm hoping to use vanilla js numbers wherever possible to avoid issues like those presented in the sharding gitter.

One example of pain from using bignums: Look at our `ValidatorIndex` type, javascript doesn't support `Array`s with length greater than 2**52, but `ValidatorIndex` can theoretically go to ~2**64. If we wanted to be the safest possible (ie, support validator lists with length > 2**53), we would need to also implement/use some alternative Array implementation ala google hugelist. (This is actually an issue regardless if we use bignums completely across the board or not :disappointed:)

Also, BNs are slow, like 30x slower than numbers. Here's a gist that shows a simple benchmark: https://gist.github.com/wemeetagain/32e94bdeedca36b3529590d99166046e

Reply:
As i posted this i got into a conversation on the gitter... 

> If we wanted to be the safest possible (ie, support validator lists with length > 2**53), we would need to also implement/use some alternative Array implementation ala google hugelist. (This is actually an issue regardless if we use bignums completely across the board or not 😞)

Note taken ...